c/src/nnstreamer-capi-pipeline.c

```diff
@@ -1076,6 +1076,38 @@ ml_pipeline_stop (ml_pipeline_h pipe)
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief Clears all data and resets the running-time of the pipeline (more info in nnstreamer.h)
+ */
+int
+ml_pipeline_flush (ml_pipeline_h pipe, bool start)
+{
+  ml_pipeline *p = pipe;
+  int status;
+
+  check_feature_state ();
+
+  if (p == NULL)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  status = ml_pipeline_stop (pipe);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  ml_logi ("The pipeline is stopped, clear all data from the pipeline.");
+
+  /* send flush event to pipeline */
+  g_mutex_lock (&p->lock);
+  gst_element_send_event (p->element, gst_event_new_flush_start ());
+  gst_element_send_event (p->element, gst_event_new_flush_stop (TRUE));
```

Comment:
Don't you need to handle return values of ```gst_element_send_event()```?

Reply:
updated